server/plugin/util_test.go

```diff
@@ -66,6 +67,54 @@ func TestParseSubscriptionsToCommandResponse(t *testing.T) {
 	}
 }
 
+func TestGetFiltersFromServiceNow(t *testing.T) {
+	defer monkey.UnpatchAll()
+	for _, testCase := range []struct {
+		description  string
+		subscription *serializer.SubscriptionResponse
+		setupAPI     func(api *plugintest.API)
+		setupClient  func(client *mock_plugin.Client)
+	}{
+		{
+			description:  "GetFiltersFromServiceNow: with no filters",
+			subscription: testutils.GetSubscription(constants.SubscriptionTypeRecord, false),
+			setupAPI:     func(api *plugintest.API) {},
+			setupClient:  func(client *mock_plugin.Client) {},
+		},
+		{
+			description:  "GetFiltersFromServiceNow: with all filters",
+			subscription: testutils.GetSubscription(constants.SubscriptionTypeBulk, true),
+			setupAPI:     func(api *plugintest.API) {},
+			setupClient: func(client *mock_plugin.Client) {
+				client.On("SearchFilterValuesInServiceNow", testutils.GetMockArgumentsWithType("string", 4)...).Return(
+					testutils.GetServiceNowFilterValues(1), http.StatusOK, nil,
+				)
+			},
+		},
+		{
+			description:  "GetFiltersFromServiceNow: error in searching filter values",
+			subscription: testutils.GetSubscription(constants.SubscriptionTypeBulk, true),
+			setupAPI: func(api *plugintest.API) {
+				api.On("LogError", testutils.GetMockArgumentsWithType("string", 3)...)
+			},
+			setupClient: func(client *mock_plugin.Client) {
+				client.On("SearchFilterValuesInServiceNow", testutils.GetMockArgumentsWithType("string", 4)...).Return(
+					nil, http.StatusInternalServerError, fmt.Errorf("failed to get the filter values"),
+				)
+			},
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			p, api := setupTestPlugin(&plugintest.API{}, nil)
+			client := setupPluginForCheckOAuthMiddleware(p, t)
+			testCase.setupClient(client)
+			testCase.setupAPI(api)
+			defer api.AssertExpectations(t)
+			p.GetFiltersFromServiceNow(testCase.subscription, client, nil, true)
```

Comment:
what are you checking in this test case ?

Reply:
ok but we usually don't consider AssertExpectations as a test case check. There are many reasons for it:-
1) It just checks whether methods are added using API.On is called and returned as expected. But it doesn't check how many times that method is called. What is the order of the method called etc.
That's why if a function is not returning anything we check how many times a specific method is called or not.